Framework/Crystal/src/SCDCalibratePanels2ObjFunc.cpp

```diff
@@ -84,96 +99,71 @@ void SCDCalibratePanels2ObjFunc::function1D(double *out, const double *xValues,
   //-- NOTE: given that these components are never used as
   //         one vector, there is no need to construct a
   //         xValues
   UNUSED_ARG(xValues);
   UNUSED_ARG(order);
 
-  // Get workspace and component name (string type)
-  m_ws = AnalysisDataService::Instance().retrieveWS<Workspace>(
-      getAttribute("Workspace").asString());
-  m_cmpt = getAttribute("ComponentName").asString();
+  // -- always working on a copy only
+  IPeaksWorkspace_sptr pws = m_pws->clone();
 
-  // Special adjustment for CORELLI
-  PeaksWorkspace_sptr pws = std::dynamic_pointer_cast<PeaksWorkspace>(m_ws);
-  Instrument_sptr inst =
-      std::const_pointer_cast<Instrument>(pws->getInstrument());
-  if (inst->getName().compare("CORELLI") == 0 && m_cmpt != "moderator")
-    m_cmpt.append("/sixteenpack");
-
-  // NOTE: Since the feature vectors are all deltas with respect to the starting
-  // position,
-  //       we need to only operate on a copy instead of the original to avoid
-  //       changing the base value
-  std::shared_ptr<API::Workspace> calc_ws = m_ws->clone();
+  // Debugging related
+  IPeaksWorkspace_sptr pws_ref = m_pws->clone();
 
   // NOTE: when optimizing T0, a none component will be passed in.
   if (m_cmpt != "none/sixteenpack") {
     // rotation
-    // NOTE: moderator should not be reoriented
-    rotateInstrumentComponentBy(vx, vy, vz, drotang, m_cmpt, calc_ws);
+    pws = rotateInstrumentComponentBy(vx, vy, vz, drotang, m_cmpt, pws);
 
     // translation
-    moveInstruentComponentBy(dx, dy, dz, m_cmpt, calc_ws);
-  }
-
-  // generate a flatten Q_sampleframe from calculated ws (by moving instrument
-  // component) so that a direct comparison can be performed between measured
-  // and calculated
-  PeaksWorkspace_sptr calc_pws =
-      std::dynamic_pointer_cast<PeaksWorkspace>(calc_ws);
-  Instrument_sptr calc_inst =
-      std::const_pointer_cast<Instrument>(calc_pws->getInstrument());
-
-  // NOTE: We are not sure if the PeaksWorkspace level T0
-  //       if going go affect the peak.getTOF
-  Mantid::API::Run &run = calc_pws->mutableRun();
-  double T0 = 0.0;
-  if (run.hasProperty("T0")) {
-    T0 = run.getPropertyValueAsType<double>("T0");
+    pws = moveInstruentComponentBy(dx, dy, dz, m_cmpt, pws);
   }
-  T0 += dT0;
-  run.addProperty<double>("T0", T0, true);
 
-  for (int i = 0; i < calc_pws->getNumberPeaks(); ++i) {
-    const Peak pk = calc_pws->getPeak(i);
-
-    V3D hkl =
-        V3D(boost::math::iround(pk.getH()), boost::math::iround(pk.getK()),
-            boost::math::iround(pk.getL()));
-    // if (hkl == UNSET_HKL)
-    //   throw std::runtime_error("Found unindexed peak in input workspace!");
-
-    // construct the out vector (Qvectors)
+  // TODO:
+  // need to do something with dT0
+
+  // calculate residual
+  // double residual = 0.0;
+  for (int i = 0; i < pws->getNumberPeaks(); ++i) {
+    // cache TOF
+    const double tof = pws->getPeak(i).getTOF();
```

Comment:
Again, this bugs me. Who is to say that this `tof` is the best estimation of the TOF for the  peak?. As I see it, this `tof` [is obtained using L2](https://github.com/mantidproject/mantid/blob/master/Framework/DataObjects/src/Peak.cpp#L363) of the uncalibrated instrument.
A `tof` that is geometry independent should result as the average TOF of the neutron events contributing to the peak.

Reply:
Based on the information I was given, the TOF is the actually measured value, and the reset are derived based on TOF tof detector position.
There are way too many unknowns regarding how the data was translated from measured to actual peaks, and I do not believe this is the best place to start this discussion.

Reply:
I understand this is not the right place to figure this out but at least we should keep it in mind because it seems you're using this TOF and as intrinsic property of the peak, while in fact it's been calculated using the instrument and hence is sensitive to the calibration, or lack of

Reply:
Frankly I am getting mixed information regarding what quantity is the actual measured value, and I am not sure what I can do at this point.